apps/openmw/mwrender/actoranimation.cpp

```diff
@@ -51,6 +63,302 @@ ActorAnimation::~ActorAnimation()
     {
         mInsert->removeChild(iter->second);
     }
+
+    mScabbard.reset();
+}
+
+PartHolderPtr ActorAnimation::getWeaponPart(const std::string& model, const std::string& bonename, bool enchantedGlow, osg::Vec4f* glowColor)
+{
+    osg::Group* parent = getBoneByName(bonename);
+    if (!parent)
+        return nullptr;
+
+    osg::ref_ptr<osg::Node> instance = mResourceSystem->getSceneManager()->getInstance(model, parent);
+
+    const NodeMap& nodeMap = getNodeMap();
+    NodeMap::const_iterator found = nodeMap.find(Misc::StringUtils::lowerCase(bonename));
+    if (found == nodeMap.end())
+        return PartHolderPtr();
+
+    if (enchantedGlow)
+        addGlow(instance, *glowColor);
+
+    return PartHolderPtr(new PartHolder(instance));
+}
+
+osg::Group* ActorAnimation::getBoneByName(std::string boneName)
+{
+    if (!mObjectRoot)
+        return nullptr;
+
+    SceneUtil::FindByNameVisitor findVisitor (boneName);
+    mObjectRoot->accept(findVisitor);
+
+    return findVisitor.mFoundNode;
+}
+
+std::string ActorAnimation::getHolsteredWeaponBoneName(const MWWorld::ConstPtr& weapon)
+{
+    std::string boneName = "";
```

Comment:
You don't need to assign a string to "" to keep it empty. Simply declare it like `std::string boneName;`. That will be a little faster and cleaner. Same thing in the getHolsteredWeaponBoneName(weaponType).

Reply:
We use the same approach in many other places. If we want to change it, I suppose we need to replace other usages too (better as a separate PR).

Reply:
Zini would have vetoed such a PR for very good reasons.

Reply:
OK, I replaced it, but it does not make situation better - we still widely "" in OpenMW code.

Reply:
We've done clean-ups in the past in PRs, like the push to C++11. 

I believe that whole-sale cleanups, like code re-formating, would be rejected because it adds nothing and masks git history. 

So a separate PR simply dropping empty string decelerations is in between the two, however in the grand scheme of things isn't that bad but more welcome.